pallets/cash/src/internal/initialize.rs

```diff
@@ -115,12 +115,17 @@ pub fn initialize_block<T: Config>(now: Timestamp) -> Result<(), Reason> {
         }
     }
 
-    <Module<T>>::deposit_event(Event::TransferCash(
-        GATEWAY_VOID,
-        last_miner,
-        last_miner_share_principal,
-        cash_index_new,
-    ));
+    if last_miner_share_principal != CashPrincipalAmount::ZERO {
+        // Filter Transfer 0 events emitted when miners are paid
```

Comment:
Maybe just "No need to emit events when nothing happens", these wont be cluttering anything up once this is merged